src/ui/components/AliasSelector.js

```diff
@@ -0,0 +1,162 @@
+// @flow
+
+import React, {useState, useMemo} from "react";
+import {useCombobox, useMultipleSelection} from "downshift";
+import {Ledger} from "../../ledger/ledger";
+import {type Identity} from "../../ledger/identity";
+import {CredView} from "../../analysis/credView";
+import {type NodeAddressT} from "../../core/graph";
+import Markdown from "react-markdown";
+
+type Props = {|
+  +currentIdentity: Identity | null,
+  +ledger: Ledger,
+  +credView: CredView,
+  +setLedger: (Ledger) => void,
+  +setCurrentIdentity: (Identity) => void,
+|};
+
+export function AliasSelector({
+  currentIdentity,
+  ledger,
+  setLedger,
+  setCurrentIdentity,
+  credView,
+}: Props) {
+  const [inputValue, setInputValue] = useState("");
+  const {
+    getSelectedItemProps,
+    getDropdownProps,
+    addSelectedItem,
+    removeSelectedItem,
+    selectedItems,
+  } = useMultipleSelection({
+    initialSelectedItems: [],
+  });
+  useMemo(() => {
+    selectedItems.forEach((item) => {
+      removeSelectedItem(item);
+    });
+    if (currentIdentity) {
+      currentIdentity.aliases.forEach((aliasAddress) =>
+        addSelectedItem(aliasAddress)
+      );
+    }
+  }, [currentIdentity]);
+
+  const getNodeDescription = (nodeAddress: NodeAddressT): string => {
+    const node = credView.node(nodeAddress);
+    if (node) {
+      return node.description;
+    }
+    return "";
+  };
+
+  const getFilteredItems = (items) =>
+    items.filter(
+      (item) =>
+        !ledger._aliases.has(item) &&
```

Comment:
didn't catch this in initial review, but: if a property is `_named` like `ledger._aliases`, then it's a module-internal implementation detail, and it shouldn't be accessed externally. the right solution is either to to get the same data via existing apis (`ledger.accounts().map((a) => a.identity.aliases)` in this case) or to add a new API. Ill fix this up.